### PR TITLE
Refactor: deprecated global const variables in favor of configurable functional options

### PR DESCRIPTION
**Pull request by derrandz**
# Overview

The codebase is currently littered with default constant global variables acting as configuration variables.
This makes changing configuration values for the Celestia node non-trivial, as it requires code changes and binary re-building.

In this PR, we refactor the global configuration variables logic to use the functional options pattern, which will trivialize the process of configuring a Celestia node.

# References

#709 
#1063 
#912 

## Impact

- [x] the `share` module
- [ ] the `nodebuilder` module
- [x] the `das` module
- [ ] the `fraud` module
- [ ] the `header` module
- [ ] the `ipld` module
- [ ] the `params` module
- [ ] the `rpc` module

### CODE REVIEW

**Review by Wondertan — COMMENTED** [review]
@derrandz, I thought you wanted to start from DASer 😅 . There is ongoing refactoring in share pkg #1177, and it will be easier to work on share pkg afterward. 

Also, you made the Impact todo list of all the pkgs within this PR, but we should go about this with 1pkg:1PR, so I would rebrand this pr to be specifically for the `share` pkg and leave it until the #1177 is merged. Meanwhile, you can work on other pkgs. 

It will be useful to transfer the Impact list on the issue and track progress of each pkg there

File: Makefile [hunk 1]
```diff
@@ -28,6 +28,11 @@ clean:
 	@echo "--> Cleaning up ./build"
 	@rm -rf build/*
 
+## clean-cache: removes all go cache, useful for when your tests aren't passing due to cached old code
```
Comment (Wondertan) [inline Makefile:31]: Not sure I get the use case. How tests cannot pass because of old cached code? I've never encountered this. Maybe there is another solution to the problem?

File: Makefile [hunk 2]
```diff
@@ -28,6 +28,11 @@ clean:
...
+clean-cache:
+	@go clean -r -testcache -cache -modcache .
+.PHONY: clean-cache
```
Comment (Wondertan) [inline Makefile:34]: This should not be part of the PR. Pls extract it into a separate one

File: Makefile (hunk 1 above)
Comment (derrandz) [reply Makefile:31]: Prior to go1.18, I've used to run into issues such as not seeing a debug line I've put in a test or in a module being tested, and pruning the cache made those lines appear, and while I was making my way with this PR, I had to run this line to remove any doubts.

I agree, the use-case is not well-defined. (I don't feel strongly about this addition)

Comment (derrandz) [reply Makefile:31]: Actually, I've just ran into this with the `das` package. Maybe we can discuss this and ship it in another PR.

**Comment by derrandz** [comment]
> @derrandz, I thought you wanted to start from DASer 😅 . There is ongoing refactoring in share pkg #1177, and it will be easier to work on share pkg afterward.
> 
> Also, you made the Impact todo list of all the pkgs within this PR, but we should go about this with 1pkg:1PR, so I would rebrand this pr to be specifically for the `share` pkg and leave it until the #1177 is merged. Meanwhile, you can work on other pkgs.
> 
> It will be useful to transfer the Impact list on the issue and track progress of each pkg there

@Wondertan Thank you for eyeballing this WIP, and thanks for raising my attention to the on-going refactors (_I wasn't aware_). I agree with the rebranding suggestion, I will close this PR in favor of PRs of other modules (_the 1pkg:1pr_)

So far, this PR has only impacted:

- `share` package
- `das` package

**Comment by derrandz** [comment]
Closing in favor of #1224 #1225 as per the conversation above.
